pkg/client/auditor/auditor.go

```diff
@@ -392,8 +392,17 @@ func (a *defaultAuditor) verifyStateSignature(
 	if serverState.GetSignature() != nil {
 		pk := a.serverSigningPubKey
 		if pk == nil {
-			x, y := elliptic.Unmarshal(elliptic.P256(), serverState.GetSignature().GetPublicKey())
-			pk = &ecdsa.PublicKey{Curve: elliptic.P256(), X: x, Y: y}
+			a.logger.Warningf(
+				"server signature will be verified using untrusted public key (embedded in the server state payload) " +
+					"- for better security please configure a public key for the auditor process")
+			var err error
+			pk, err = signer.UnmarshalKey(serverState.GetSignature().GetPublicKey())
+			if err != nil {
+				return fmt.Errorf(
+					"failed to verify signature for state %s at TX %d received from server %s @ %s: "+
+						"error unmarshaling the public key embedded in the server state payload: %v",
```

Comment:
@byo Indeed: i use `%w` when it makes sense for the caller to check the error type with `errors.Is`, otherwise - e.g. if the caller only logs the error, like in this case - i see no point in wrapping it. Unless it is considered a good practice to wrap each and every error returned. In this case we should replace each and every occurrence of `%v` for errors with `%w` throughout the whole codebase.

Reply:
Ah, true, forgot that this is within the auditor binary itself. Leaving the decision up to you then - maybe better to stay consistent with the rest of auditor's code?